### PR TITLE
Add option to pull externally linked account details

### PR DESCRIPTION
**Pull request by sbose78**
Add support for 
`/api/token?force_pull=true&for=https://github.com`

### CODE REVIEW

**Comment by fabric8cd** [comment]
@sbose78 snapshot fabric8-auth image is available for testing.  `docker pull docker.io/fabric8/fabric8-auth:SNAPSHOT-PR-218-1`

**Comment by sbose78** [comment]
@alexeykazakov let me know your thoughts on having something like this. 
This piece can be used to even test out whether the token is valid.

File: controller/token.go [hunk 1]
```diff
@@ -250,11 +252,14 @@ func (c *TokenController) Retrieve(ctx *app.RetrieveTokenContext) error {
 		return jsonapi.JSONErrorResponse(ctx, err)
 	}
 	if externalToken != nil {
-		updatedToken, err := c.updateProfileIfEmpty(ctx, providerConfig, externalToken)
-		if err != nil {
-			return jsonapi.JSONErrorResponse(ctx, err)
+		if externalToken.Username == "" || (ctx.ForcePull != nil && *ctx.ForcePull == true) {
```
Comment (alexeykazakov) [inline controller/token.go]: 1. `if *ctx.ForcePull == true {...}` looks weird :) `if *ctx.ForcePull {...}` would look better ;)
2. Let's move the check of this param to the `updateProfileIfEmpty(ctx, config, token, force)` method?

File: controller/token_storage_blackbox_test.go [hunk 2]
```diff
@@ -93,7 +93,7 @@ func (rest *TestTokenStorageREST) checkRetrieveOSOServiceAccountToken(saName str
 	service, controller := rest.SecuredControllerWithServiceAccount(sa)
 	require.True(rest.T(), len(rest.Configuration.GetOSOClusters()) > 0)
 	for _, cluster := range rest.Configuration.GetOSOClusters() {
-		_, tokenResponse := test.RetrieveTokenOK(rest.T(), service.Context, service, controller, cluster.URL)
+		_, tokenResponse := test.RetrieveTokenOK(rest.T(), service.Context, service, controller, cluster.URL, nil)
```
Comment (alexeykazakov) [inline controller/token_storage_blackbox_test.go]: Add some tests when the force param is not nil and handled properly.

File: design/token.go [hunk 3]
```diff
@@ -38,6 +38,7 @@ var _ = a.Resource("token", func() {
 		)
 		a.Params(func() {
 			a.Param("for", d.String, "The resource for which the external token is being fetched, example https://github.com/fabric8-services/fabric8-auth or https://api.starter-us-east-2.openshift.com")
+			a.Param("force_pull", d.Boolean, "Pull the user's details for the specific connected account, example, the user's updated github username would be fetched from github")
```
Comment (alexeykazakov) [inline design/token.go]: Mention that if it's not set or "false" then the user profile will be pulled only if it's missing in Auth. Otherwise the profile stored in Auth will be used instead of updating it from the provider.

**Comment by fabric8cd** [comment]
@sbose78 snapshot fabric8-auth image is available for testing.  `docker pull docker.io/fabric8/fabric8-auth:SNAPSHOT-PR-218-2`

File: design/token.go (hunk 3 above)
Comment (sbose78) [reply design/token.go]: ad55329

Done.

File: controller/token_storage_blackbox_test.go (hunk 2 above)
Comment (sbose78) [reply controller/token_storage_blackbox_test.go]: Added 2 tests:
06656cecf2653d2e3d7fdf6d064543083f9a0323

1. Token present in DB, but error during interaction with github.com
2. Token present in DB, & success during interaction with github.com

File: design/token.go (hunk 3 above)
Comment (sbose78) [reply design/token.go]: Done in 06656cecf2653d2e3d7fdf6d064543083f9a0323

**Comment by fabric8cd** [comment]
@sbose78 snapshot fabric8-auth image is available for testing.  `docker pull docker.io/fabric8/fabric8-auth:SNAPSHOT-PR-218-3`
